### PR TITLE
Expand readme for configuration parameters

### PR DESCRIPTION
**Pull request by Steveb-p**
See https://github.com/arnaud-lb/php-rdkafka/issues/305#issuecomment-547322127

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,7 +162,7 @@ Creating the queue:
 $queue = $rk->newQueue();
 ```
 
-Adding topars to the queue:
+Adding topics to the queue:
```
Comment (nick-zh) [inline README.md]:
```suggestion
Adding topic partitions to the queue:
```

File: README.md [hunk 2]
```diff
@@ -196,12 +196,25 @@ while (true) {
 
 ### Using stored offsets
 
-librdkafka can store offsets in a local file, or on the broker. The default is local file, and as soon as you start using ``RD_KAFKA_OFFSET_STORED`` as consuming offset, rdkafka starts to store the offset.
+librdkafka can store offsets in a local file, or on the broker.
```
Comment (nick-zh) [inline README.md]:
```suggestion
#### Broker (default)
librdkafka per default stores offsets on the broker.
```

File: README.md [hunk 3]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
 
-By default, the file is created in the current directory, with a name based on the topic and the partition. The directory can be changed by setting the ``offset.store.path`` [configuration property](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
+For older versions of librdkafka library, default value was storing offset values in local file, but it has since been
```
Comment (nick-zh) [inline README.md]:
```suggestion
#### File offsets (deprecated)
```
we probably shouldn't give guidance for a deprecated feature (imho)

File: README.md [hunk 4]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
+deprecated. Currently by default librdkafka will use broker as offset storage. You can switch it back by using
```
Comment (nick-zh) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
+`offset.store.method` configuration property.
```
Comment (nick-zh) [inline README.md]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
+
+If you're using local file for offset storage, then by default, the file is created in the current directory, with a
+name based on the topic and the partition. The directory can be changed by setting the ``offset.store.path``
+[configuration property](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
 
 Other interesting properties are: ``auto.commit.interval.ms``, ``auto.commit.enable``, ``group.id``, ``max.poll.interval.ms``.
```
Comment (nick-zh) [inline README.md:210]:
```suggestion
#### Useful offset settings
Other interesting properties are: ``auto.commit.interval.ms``, ``auto.commit.enable``, ``group.id``, ``max.poll.interval.ms``.
```

File: README.md [hunk 7]
```diff
@@ -284,7 +297,7 @@ while ($producer->getOutQLen() > 0) {
 
 ## Documentation
 
-https://arnaud-lb.github.io/php-rdkafka/phpdoc/book.rdkafka.html  
+https://arnaud-lb.github.io/php-rdkafka/phpdoc/book.rdkafka.html
```
Comment (nick-zh) [inline README.md]:
```suggestion
https://arnaud-lb.github.io/php-rdkafka/phpdoc/book.rdkafka.html  
```

**Review by nick-zh — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
+If you're using local file for offset storage, then by default, the file is created in the current directory, with a
+name based on the topic and the partition. The directory can be changed by setting the ``offset.store.path``
+[configuration property](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
+
+#### Useful offset settings
 Other interesting properties are: ``auto.commit.interval.ms``, ``auto.commit.enable``, ``group.id``, ``max.poll.interval.ms``.
 
+`auto.commit.interval.ms` and `auto.commit.enable` work in tandem: unless you specify otherwise, consumers **WILL**
+commit automatically in the background (at least high-level ones). If you need control and want to commit manually,
+then you want to disable `auto.commit.enable`.
```
Comment (nick-zh) [inline README.md]:
```suggestion
then you want to set `auto.commit.enable=false`.
```

**Review by nick-zh — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -196,12 +196,25 @@ while (true) {
...
+
+`group.id` is responsible for setting your consumer group ID and it should be unique for your application (and should
```
Comment (nick-zh) [inline README.md]:
```suggestion
`group.id` is responsible for setting your consumer group ID and it should be unique (and should
```

**Review by nick-zh — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (Steveb-p) [reply README.md]: Replaced with
```md
to set `auto.commit.enable` to `'false'`
```
which should be even more accurate I believe?

File: README.md (hunk 9 above)
Comment (Steveb-p) [reply README.md]: Removed accordingly.

**Review by nick-zh — APPROVED** [review]

**Comment by nick-zh** [comment]
Thanks @Steveb-p

**Pull request merged**